File: 2024/JavaScript/Challenge#2.js

function createFrame(names) {
    const largest_name = names.reduce((prev, curr) => prev.length > curr.length ? prev : curr, "");
    const largest_length = largest_name.length;

    let frame = '*'.repeat(largest_length + 4) + "\n";
    for (const name of names) {
        frame += "* " + name.padEnd(largest_length) + " *\n";
    }
    frame += '*'.repeat(largest_length + 4);

    return frame;
}


const frame1 = createFrame(['midu', 'madeval', 'educalvolpz']);
console.log(frame1);
// Expected result:
/*
***************
* midu        *
* madeval     *
* educalvolpz *
***************
*/

const frame2 = createFrame(['midu']);
console.log(frame2);
// Expected result:
/*
********
* midu *
********
*/

const frame3 = createFrame(['a', 'bb', 'ccc']);
console.log(frame3);
// Expected result:
/*
*******
* a   *
* bb  *
* ccc *
*******
*/

File: 2024/Python/Challenge#2.py
from typing import List

def create_frame(names: List[str]) -> str:
    # largest_name = max (names, key=len) if names else ""
    largest_name = ""
    for name in names:
        if len(name) > len(largest_name):
            largest_name = name
    largest_length = len(largest_name)
    
    frame = '*' * (largest_length + 4) + "\n"
    for name in names:
        frame += "* " + name.ljust(largest_length) + " *\n"
    frame += '*' * (largest_length + 4)
    
    return frame


frame1 = create_frame(['midu', 'madeval', 'educalvolpz'])
print(frame1)
# Expected result:
'''
***************
* midu        *
* madeval     *
* educalvolpz *
***************
'''

frame2 = create_frame(['midu'])
print(frame2)
# Expected result:
'''
********
* midu *
********
'''

frame3 = create_frame(['a', 'bb', 'ccc'])
print(frame3)
# Expected result:
'''
*******
* a   *
* bb  *
* ccc *
*******
'''
